src/test/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveStreamCommandsIntegrationTests.java

```diff
@@ -340,6 +342,148 @@ void xPendingShouldLoadPendingMessagesForNonExistingConsumer() {
 				}).verifyComplete();
 	}
 
+	@ParameterizedRedisTest // GH-2046
+	void xPendingShouldLoadPendingMessagesForGroupAndIdle() {
+
+		String initialMessage = nativeCommands.xadd(KEY_1, KEY_1, VALUE_1);
+		nativeCommands.xgroupCreate(XReadArgs.StreamOffset.from(KEY_1, initialMessage), "my-group");
+
+		nativeCommands.xadd(KEY_1, KEY_2, VALUE_2);
+
+		connection.streamCommands()
+				.xReadGroup(Consumer.from("my-group", "my-consumer"),
+						StreamOffset.create(KEY_1_BBUFFER, ReadOffset.lastConsumed()))
+				.then().as(StepVerifier::create).verifyComplete();
+
+		Duration exceededIdle = Duration.of(1, ChronoUnit.MILLIS);
+
+		connection.streamCommands().xPending(KEY_1_BBUFFER, "my-group", Range.open("-", "+"), 10L, exceededIdle)
+				.as(StepVerifier::create).assertNext(it -> {
+					assertThat(it.size()).isOne();
+					assertThat(it.get(0).getConsumerName()).isEqualTo("my-consumer");
+					assertThat(it.get(0).getGroupName()).isEqualTo("my-group");
+					assertThat(it.get(0).getTotalDeliveryCount()).isOne();
+					assertThat(it.get(0).getIdAsString()).isNotNull();
+				}).verifyComplete();
```

Comment:
this construct caused some flakiness when running tests had to `delaySubscription(...)` on `xPending` to get it stable.